pandas/plotting/_core.py

```diff
@@ -375,7 +375,7 @@ def boxplot(
         >>> type(boxplot)
         <class 'numpy.ndarray'>
     """
-    plot_backend = _get_plot_backend()
+    plot_backend = _get_plot_backend("matplotlib")
```

Comment:
Should we instead raise?

```python
pd.options.plotting.backend = "altair"
df.boxplot()
```

That would raise an exception saying to use `DataFrame.plot(kind="box")` or set the backend to matplotlib.

Reply:
I think this function is `pandas.plotting.boxplot(df)` and not `df.boxplot()` which iirc is `boxplot_frame`, and in this PR still delegates the call to the backend.

Personally I'd deprecate `pandas.plotting.boxplot` more than anything else. Afaik it's the same as `df.boxplot()` or `df.plot(kind='box')`, each with a different implementation. But if we don't deprecate, my preference would be to try to make things as simple as possible, and avoid special cases.